src/engine/cache/file-cache.js

```diff
@@ -231,8 +248,13 @@ class FileCache extends BaseCache {
    * @returns {Promise<void>} - The result promise
    */
   async removeFileIfTooOld(filePath, referenceDate, archiveFolder) {
-    const stats = await fs.stat(path.join(archiveFolder, filePath))
-    if (stats.mtimeMs + this.retentionDuration < referenceDate) {
+    let stats
+    try {
+      stats = await fs.stat(path.join(archiveFolder, filePath))
+    } catch (error) {
+      this.logger.error(error)
```

Comment:
a comment explaining why this can happen and why silently ignoring it is the right way to go would be more than helpful